### PR TITLE
Add recommended Helm labels to manifest charts

### LINKED ISSUE #25992: Update Istio Helm charts to enable Helm adoption of existing resources
**Describe the feature request**

Several Istio users that I work with, including myself, still use Helm to manage Istio deployments. A frequent problem that we run into after failed installations is Helm complaining about resources that already exist and refusing to adopt or overwrite them (understandably). There can be a lot of root causes...maybe something is wrong in a values file, or you forgot to set an option that you wanted/used the wrong values file. Maybe because of it, one component didn't come up healthy, so the Helm install errored out and is in a "failed" status.

In this case, you would just want to correct your values file and do another install over the top of the failed one to fix it. Now, because Helm doesn't adopt those failed resources, you may have orphans. You now have to completely uninstall everything and try again, or delete the offending resources one by one if you can't uninstall completely.

This has always been an issue with Helm in general with being unable to adopt resources. Well, Helm 3 finally fixed it using a certain set of annotations and labels in the deployed resources. See here: https://github.com/helm/helm/pull/7649. It doesn't work with hook resources at the moment (likely a bug/oversight), but seems to work well for everything else.

Adding these annotations/labels to Istio's Helm charts to allow Helm to adopt resources left behind by failed (and other) deployments would be beneficial to us, and make the deployment/upgrade experience much smoother for users.

**Describe alternatives you've considered**

Adding hooks and hook-delete-policies to all resources, but this is something users would have to do after the fact in their own fork of Istio. This is also something that would be better implemented within the Istio project itself, rather than users having to edit and maintain their own charts. I believe the requested solution leveraging the new Helm functionality is more elegant.

[X] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

**Additional context**

### PR DESCRIPTION
**Pull request by jacob-delgado**
Fixes https://github.com/istio/istio/issues/25992

Following Helm best practices as discussed in https://helm.sh/docs/chart_best_practices/labels/

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by howardjohn** [comment]
The upgrade note suggests this actually changes the behavior of helm. is that the case?

**Comment by jacob-delgado** [comment]
> The upgrade note suggests this actually changes the behavior of helm. is that the case?

It should only change the behavior for failed upgrades; where they can break when attempting to fix a bad install. This should help it become more resilient. But I also need to verify what happens if you add labels to an already existing installation. It shouldn't break anything if I read helm's documentation correctly.

I am concerned about the behavior of istioctl with these labels. We don't support going from istioctl to helm (or vice versa) so it needs to be seen that this doesn't change the behavior. We may need to somehow guard these against them in istioctl. Not sure if such behavior already exist.

File: releasenotes/notes/25992.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# issue is a list of GitHub issues resolved in this note.
+issue:
+  - https://github.com/istio/istio/issues/25992
+
+releaseNotes:
+  - |
+    **Fixed** failed Helm installations not adopting existing resources
+
+upgradeNotes:
+  - title: Helm chart now use recommended labels
```
Comment (ericvn) [inline releasenotes/notes/25992.yaml:14]: Should be `Helm charts now use` or `Helm chart now uses`

**Comment by howardjohn** [comment]
Which labels specifically does it need? All of them?

They are quite a pain to have a bunch of labels in my experience. We
intentionally removed them a while back

File: releasenotes/notes/25992.yaml (hunk 1 above)
Comment (jacob-delgado) [reply releasenotes/notes/25992.yaml]: fixed

**Comment by jacob-delgado** [comment]
> Which labels specifically does it need? All of them? They are quite a pain to have a bunch of labels in my experience. We intentionally removed them a while back

Here's the documentation from helm and k8s

| annotation                   | K8s docs                                                      | Helm docs                                                                                                                                                                      | Required to fix |
|------------------------------|---------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|
| app.kubernetes.io/name       | The name of the application                                   | This should be the app name, reflecting the entire app. Usually {{ template "name" . }} is used for this. This is used by many Kubernetes manifests, and is not Helm-specific. | No              |
| app.kubernetes.io/managed-by | The tool being used to manage the operation of an application | This should always be set to {{ .Release.Service }}. It is for finding all things managed by Helm.                                                                             | Yes             |
| app.kubernetes.io/instance   | A unique name identifying the instance of an application      | This should be the {{ .Release.Name }}. It aids in differentiating between different instances of the same application.                                                        | No              |
| helm.sh/chart                | N/A                                                           | This should be the chart name and version:   {{ .Chart.Name }}-{{ .Chart.Version \| replace "+" "_" }}.                                                                        |                 |

When using `helm create` it creates a template with:
* helm.sh/chart
* app.kubernetes.io/name
* app.kubernetes.io/instance
* app.kubneretes.io/managed-by
and if `appVersion` is specified also `app.kubernetes.io/version`.

I've changed it a bit to work with the templates as done by `helm create` so I can easily change it.

Judging by what I've read, we only need `app.kubernetes.io/managed-by` by judging https://github.com/helm/helm/blob/ab12382ddcd3a27a974b45ed96e47697de221244/pkg/action/validate.go#L80

**Pull request closed without merging**